diff --git a/src/controller/CurrentFabricRemover.cpp b/src/controller/CurrentFabricRemover.cpp
index 805504cb3..80a573f9a 100644
--- a/src/controller/CurrentFabricRemover.cpp
+++ b/src/controller/CurrentFabricRemover.cpp
@@ -150,6 +150,7 @@ void CurrentFabricRemover::FinishRemoveCurrentFabric(void * context, CHIP_ERROR
     self->mNextStep = Step::kAcceptRemoveFabricStart;
     if (self->mCurrentFabricRemoveCallback != nullptr)
     {
+        ChipLogError(Controller, "Sending out the current fabric remove callback. should trigger python callback");
         self->mCurrentFabricRemoveCallback->mCall(self->mCurrentFabricRemoveCallback->mContext, self->mRemoteNodeId, err);
     }
 }
diff --git a/src/controller/python/ChipDeviceController-ScriptBinding.cpp b/src/controller/python/ChipDeviceController-ScriptBinding.cpp
index eaced9431..4532ea000 100644
--- a/src/controller/python/ChipDeviceController-ScriptBinding.cpp
+++ b/src/controller/python/ChipDeviceController-ScriptBinding.cpp
@@ -48,6 +48,7 @@
 #include <controller/CHIPDeviceControllerFactory.h>
 #include <controller/CommissioningDelegate.h>
 #include <controller/CommissioningWindowOpener.h>
+#include <controller/CurrentFabricRemover.h>
 #include <controller/ExampleOperationalCredentialsIssuer.h>
 
 #include <controller/python/ChipDeviceController-ScriptDevicePairingDelegate.h>
@@ -89,6 +90,7 @@ typedef void (*ConstructBytesArrayFunct)(const uint8_t * dataBuf, uint32_t dataL
 typedef void (*LogMessageFunct)(uint64_t time, uint64_t timeUS, const char * moduleName, uint8_t category, const char * msg);
 typedef void (*DeviceAvailableFunc)(DeviceProxy * device, PyChipError err);
 typedef void (*ChipThreadTaskRunnerFunct)(intptr_t context);
+typedef void (*DeviceUnpairingCompleteFunct)(uint64_t nodeId, CHIP_ERROR error);
 }
 
 namespace {
@@ -131,6 +133,7 @@ PyChipError pychip_DeviceController_ConnectIP(chip::Controller::DeviceCommission
                                               uint32_t setupPINCode, chip::NodeId nodeid);
 PyChipError pychip_DeviceController_ConnectWithCode(chip::Controller::DeviceCommissioner * devCtrl, const char * onboardingPayload,
                                                     chip::NodeId nodeid);
+PyChipError pychip_DeviceController_UnpairDevice(chip::Controller::DeviceCommissioner * devCtrl, chip::NodeId remoteDeviceId, DeviceUnpairingCompleteFunct callback);
 PyChipError pychip_DeviceController_SetThreadOperationalDataset(const char * threadOperationalDataset, uint32_t size);
 PyChipError pychip_DeviceController_SetWiFiCredentials(const char * ssid, const char * credentials);
 PyChipError pychip_DeviceController_CloseSession(chip::Controller::DeviceCommissioner * devCtrl, chip::NodeId nodeid);
@@ -386,6 +389,28 @@ PyChipError pychip_DeviceController_ConnectWithCode(chip::Controller::DeviceComm
     return ToPyChipError(devCtrl->PairDevice(nodeid, onboardingPayload, sCommissioningParameters));
 }
 
+PyChipError pychip_DeviceController_UnpairDevice(chip::Controller::DeviceCommissioner * devCtrl, chip::NodeId nodeid, DeviceUnpairingCompleteFunct callback)
+{
+    // Create a new CurrentFabricRemover instance
+    auto fabricRemover = new chip::Controller::CurrentFabricRemover(devCtrl);
+
+    auto * callbacks = new UnpairDeviceCallbacks(callback, fabricRemover);
+    
+    // Pass the callback and nodeid to the RemoveCurrentFabric function
+    CHIP_ERROR err = fabricRemover->RemoveCurrentFabric(nodeid, &callbacks->mOnCurrentFabricRemove);
+
+    // Clean up the callback and fabricRemover object if an error occurs
+    if (err != CHIP_NO_ERROR)
+    {
+        delete fabricRemover; 
+        delete callbacks;     
+    }
+
+    ChipLogError(Controller, "Sent out Unpair command with no issues...");
+    return ToPyChipError(err);
+}
+
+
 PyChipError pychip_DeviceController_OnNetworkCommission(chip::Controller::DeviceCommissioner * devCtrl, uint64_t nodeId,
                                                         uint32_t setupPasscode, const uint8_t filterType, const char * filterParam)
 {
@@ -629,6 +654,25 @@ struct GetDeviceCallbacks
     Callback::Callback<OnDeviceConnectionFailure> mOnFailure;
     DeviceAvailableFunc mCallback;
 };
+
+struct UnpairDeviceCallbacks
+{
+    UnpairDeviceCallbacks(DeviceUnpairingCompleteFunct callback, chip::Controller::CurrentFabricRemover remover) :
+        mCallback(callback), mRemover(remover), mOnCurrentFabricRemove(OnCurrentFabricRemoveFn, this) 
+     {}
+
+    static void OnCurrentFabricRemoveFn(void * context, chip::NodeId nodeId, CHIP_ERROR error)
+    {
+        auto * self = static_cast<UnpairDeviceCallbacks *>(context);
+        self->mCallback(nodeId, ToPyChipError(error));
+        delete self->mRemover;
+        delete self;
+    }
+
+    Callback::Callback<OnCurrentFabricRemove> mOnCurrentFabricRemove;
+    DeviceUnpairingCompleteFunct mCallback;
+    chip::Controller::CurrentFabricRemover mRemover;
+};
 } // anonymous namespace
 
 PyChipError pychip_GetConnectedDeviceByNodeId(chip::Controller::DeviceCommissioner * devCtrl, chip::NodeId nodeId,
diff --git a/src/controller/python/chip/ChipDeviceCtrl.py b/src/controller/python/chip/ChipDeviceCtrl.py
index 3626dac00..13eb8011f 100644
--- a/src/controller/python/chip/ChipDeviceCtrl.py
+++ b/src/controller/python/chip/ChipDeviceCtrl.py
@@ -60,6 +60,7 @@ from .native import PyChipError
 __all__ = ["ChipDeviceController"]
 
 _DevicePairingDelegate_OnPairingCompleteFunct = CFUNCTYPE(None, PyChipError)
+_DeviceUnpairingCompleteFunct = CFUNCTYPE(None, c_uint64, PyChipError)
 _DevicePairingDelegate_OnCommissioningCompleteFunct = CFUNCTYPE(
     None, c_uint64, PyChipError)
 _DevicePairingDelegate_OnOpenWindowCompleteFunct = CFUNCTYPE(
@@ -248,6 +249,17 @@ class ChipDeviceControllerBase():
             self._ChipStack.callbackRes = err
             self._ChipStack.completeEvent.set()
 
+        def HandleUnpairDeviceComplete(nodeid: int, err: PyChipError):
+            print("are we here?")
+            if err.is_success:
+                print("Succesfully unpaired device with nodeid {}".format(nodeid))
+            else:
+                print("Failed to unpair device: {}".format(err))
+            print("jesus it worked, lol")
+
+            self._ChipStack.callbackRes = err
+            self._ChipStack.completeEvent.set()
+
         def HandlePASEEstablishmentComplete(err: PyChipError):
             if not err.is_success:
                 print("Failed to establish secure session to device: {}".format(err))
@@ -286,7 +298,7 @@ class ChipDeviceControllerBase():
 
         self.state = DCState.IDLE
         self._isActive = True
-
+        self.cbHandleDeviceUnpairCompleteFunct = _DeviceUnpairingCompleteFunct(HandleUnpairDeviceComplete)
         # Validate FabricID/NodeID followed from NOC Chain
         self._fabricId = self.GetFabricIdInternal()
         self._nodeId = self.GetNodeIdInternal()
@@ -385,7 +397,15 @@ class ChipDeviceControllerBase():
             # Error 50 is a timeout
             return False
         return self._ChipStack.commissioningEventRes.is_success
+    
+    def UnpairDevice(self, nodeid: int):
+        self.CheckIsActive()
 
+        return self._ChipStack.CallAsync(
+            lambda: self._dmLib.pychip_DeviceController_UnpairDevice(
+                self.devCtrl, nodeid, self.cbHandleDeviceUnpairCompleteFunct)
+        ).raise_on_error()
+    
     def CloseBLEConnection(self):
         self.CheckIsActive()
 
@@ -1274,6 +1294,10 @@ class ChipDeviceControllerBase():
                 c_void_p, c_char_p, c_uint64]
             self._dmLib.pychip_DeviceController_ConnectWithCode.restype = PyChipError
 
+            self._dmLib.pychip_DeviceController_UnpairDevice.argtypes = [
+                c_void_p, c_uint64, _DeviceUnpairingCompleteFunct]
+            self._dmLib.pychip_DeviceController_UnpairDevice.restype = PyChipError
+
             self._dmLib.pychip_DeviceController_CloseSession.argtypes = [
                 c_void_p, c_uint64]
             self._dmLib.pychip_DeviceController_CloseSession.restype = PyChipError
